src/gibbs.jl

```diff
@@ -0,0 +1,225 @@
+"""
+    Gibbs(sampler_map::NamedTuple)
+
+An interface for block sampling in Markov Chain Monte Carlo (MCMC).
+
+Gibbs sampling is a technique for dividing complex multivariate problems into simpler subproblems.
+It allows different sampling methods to be applied to different parameters.
+"""
+struct Gibbs <: AbstractMCMC.AbstractSampler
+    sampler_map::NamedTuple
+    parameter_names::Tuple{Vararg{Symbol}}
+
+    function Gibbs(sampler_map::NamedTuple)
+        parameter_names = Tuple(keys(sampler_map))
+        return new(sampler_map, parameter_names)
+    end
+end
```

Comment:
```suggestion
struct Gibbs{NT} <: AbstractMCMC.AbstractSampler
    sampler_map::NT
end
```

`parameter_names` seems wholly unnecessary here?